src/test/java/com/xceptance/neodymium/util/SelenideAddonsTest.java

```diff
@@ -237,125 +303,107 @@ public void testWrapAssertionErrorWithoutMessage()
 
     @Test()
     @SuppressBrowsers
-    public void testSafeRunnable()
+    public void testSafeRunnableInCause()
     {
-        // preparing the test setup as kind of generator
-        AtomicInteger counter = new AtomicInteger(0);
-        List<Runnable> runArray = new ArrayList<Runnable>();
-        runArray.add(
-                     () -> {
-                         throw new StaleElementReferenceException("You shall not pass!");
-                     });
-        runArray.add(
-                     () -> {
-                         throw new StaleElementReferenceException("You shall not pass!");
-                     });
-        runArray.add(
-                     () -> {
-                         throw new StaleElementReferenceException("You shall not pass!");
-                     });
-        runArray.add(
-                     () -> {
-                         throw new StaleElementReferenceException("You shall pass!");
-                     });
-        runArray.add(
-                     () -> {
-                         throw new StaleElementReferenceException("You shall never be seen!");
-                     });
-        final Iterator<Runnable> iterator = runArray.iterator();
+        testSafe(runArrayWithSEREinCause, false);
+    }
 
-        // testing the error path after three exceptions
-        long startTime = new Date().getTime();
-        try
-        {
-            SelenideAddons.$safe(() -> {
-                counter.incrementAndGet();
-                if (iterator.hasNext())
-                {
-                    iterator.next().run();
-                }
-            });
-        }
-        catch (StaleElementReferenceException e)
-        {
-            Assert.assertTrue(e.getMessage().startsWith("You shall pass!"));
-        }
-        long endTime = new Date().getTime();
-        Assert.assertTrue(endTime - startTime > Neodymium.configuration().staleElementRetryTimeout());
-        Assert.assertEquals(counter.get(), Neodymium.configuration().staleElementRetryCount() + 1);
+    @Test()
+    @SuppressBrowsers
+    public void testSafeRunnableInMessage()
+    {
+        testSafe(runArrayWithSEREinMessage, false);
+    }
 
-        // testing the happy path after one exception
+    @Test()
+    public void testSafeSupplierInCause()
+    {
+        testSafe(runArrayWithSEREinCause, true);
+    }
+
+    @Test()
+    public void testSafeSupplierInMessage()
+    {
+        testSafe(runArrayWithSEREinMessage, true);
+    }
+
+    @Test()
```

Comment:
please move the `testIsThrowableCausedBy` and `testIsThrowableCausedByMessage` test methods before the $safe tests to  have one topic at the time.

Reply:
Done